### PR TITLE
chore: require authentication for submission

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **chore**

This PR solves [ONYX-907]

### Description

- Require authentication for submission

https://github.com/artsy/force/assets/11945712/c09805e1-258c-45a6-8f35-e7eaf77689ff

[ONYX-907]: https://artsyproduct.atlassian.net/browse/ONYX-907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/AuthDialog/AuthDialogContext.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ export type AuthDialogOptions = {
   title?: string | ((mode: AuthDialogMode) => string)
   onClose?: () => void
   onSuccess?: () => void
+  disableTapToClose?: boolean
```
Comment (MounirDhahri) [inline src/Components/AuthDialog/AuthDialogContext.tsx:43]: not sure if this is a fine approach

Comment (dzucconi) [reply src/Components/AuthDialog/AuthDialogContext.tsx:43]: Why would we want to do this?

Comment (dzucconi) [reply src/Components/AuthDialog/AuthDialogContext.tsx:43]: IMO, it's annoying and you'll be able to close via escape or the close button anyway. If we want a required auth step then redirect to the full page login/sign up.

Comment (MounirDhahri) [reply src/Components/AuthDialog/AuthDialogContext.tsx:43]: True, didn't think about keyboard events - I will tell Barney about it. Thanks @dzucconi 🙏

Comment (MounirDhahri) [reply src/Components/AuthDialog/AuthDialogContext.tsx:43]: Done now - added a redirect to the landing page if the user is not logged in

**Review by tam-kis — APPROVED** [review]

**Pull request merged**